Asymmetric Sync (#7880)
realm/realm-swift: CHANGELOG.md

x.y.z Release notes (yyyy-MM-dd)
=============================================================
### Enhancements

> * Add Support for asymmetric sync. When an object inherit from `AsymmetricObject` is immediately
>   marked as asymmetric, upon creation this objects is sync'd unidirectionally and cannot be query
>   locally.

Reviewer: ```suggestion
* Add support for asymmetric sync. When a class inherits from
  `AsymmetricObject`, objects created are synced unidirectionally to the server
  and cannot be queried or read locally.
```

All the "sync'd" all over the place should be "synced".

  
```swift
    class PersonObject: AsymmetricObject {

>        @Persisted(primaryKey: true) var _id: ObjectId = ObjectId.generate()

Reviewer: ```suggestion
       @Persisted(primaryKey: true) var _id: ObjectId
```

       @Persisted var name: String
       @Persisted var age: Int
    }
    
    try realm.write {
       let person = PersonObject(name: "Dylan", age: 20, isFavorite: true)

>        realm.create(person) // This will create the object and sync'd unidirectionally.

Reviewer: ```suggestion
        // This will create the object on the server but not locally
       realm.create(person) 
```
